### PR TITLE
Implemented texture_2D_layered_initialize

### PR DESCRIPTION
**Pull request by patrickdown**
This change implements functionality for texture_2D_layered_initialize that was previously stubbed for OpenGL.

### CODE REVIEW

**Review by BastiaanOlij — APPROVED** [review]
This looks good to me, @clayjohn might want to give a second opinion in case we've missing something but I think this is correct.

**Review by clayjohn — COMMENTED** [review]
This looks mostly good, but it should include a bit more validation like what is done in the RD implementation of this function:
https://github.com/godotengine/godot/blob/master/servers/rendering/renderer_rd/storage_rd/texture_storage.cpp#L859

Particularly, this is missing support for when the texture type is TEXTURE_LAYERED_CUBEMAP

**Review by clayjohn — COMMENTED** [review]
Looks mostly good. I just made a small suggestion for the error

**Comment by clayjohn** [comment]
Looks great! The final step before merging is to squash all the commits together so that the whole PR only contains 1 big commit with all your changes. We like to merge one commit at a time to keep the git history clean and navigable. 

If you don't know how to do that, we have a helpful tutorial in the official documentation https://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#the-interactive-rebase

**Review by akien-mga — APPROVED** [review]
Confirming @clayjohn's implicit approval now that it has been squashed.

**Comment by akien-mga** [comment]
Thanks! And congrats for your first merged Godot contribution :tada:

**Pull request merged**
